src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1312,16 +1312,20 @@ private function dumpValue($value, $interpolate = true)
                 }
 
                 if (is_array($factory)) {
+                    $factoryMethod = substr($this->dumpValue($factory[1]), 1, -1);
+
                     if (is_string($factory[0])) {
-                        return sprintf('\\%s::%s(%s)', $factory[0], $factory[1], implode(', ', $arguments));
+                        $factoryClass = substr(str_replace('\\\\', '\\', $this->dumpValue($factory[0])), 1, -1);
```

Comment:
hmm, the factory method is indeed not resolved by the ResolveParameterPlaceHoldersPass.

and btw, the ContainerBuilder does not resolve the factoryMethod when instantiating, so using a parameter for the factoryMethod was not working consistently. It was a side effect of the dumping logic